pkg/operator/k8sutil/node.go

```diff
@@ -281,12 +281,17 @@ func GetKubernetesNodesMatchingRookNodes(ctx context.Context, rookNodes []cephv1
 	if err != nil {
 		return nodes, fmt.Errorf("failed to list kubernetes nodes. %+v", err)
 	}
-	for _, kn := range k8sNodes.Items {
-		for _, rn := range rookNodes {
+	for _, rn := range rookNodes {
+		nodeFound := false
+		for _, kn := range k8sNodes.Items {
 			if rookNodeMatchesKubernetesNode(rn, kn) {
 				nodes = append(nodes, kn)
+				nodeFound = true
```

Comment:
The previous code could have used a break, if you could add it here while you're making the change.
```suggestion
				nodeFound = true
				break
```

Reply:
Yeah I thought about that. The previous code found all the matching Kubernetes nodes so I didn't change the logic, in afraid of some nodes will be filtered out. But thinking again, maybe there should only be 1 node matches it since it's comparing node names. Will use the suggestion